src/ORM/Query.php

```diff
@@ -507,11 +507,13 @@ public function count()
         $complex = $complex || count($query->clause('union'));
 
         if (!$complex) {
+            $autoFieldsBefore = $query->eagerLoader()->autoFields();
             $query->eagerLoader()->autoFields(false);
             $statement = $query
                 ->select($count, true)
                 ->autoFields(false)
                 ->execute();
+            $query->eagerLoader()->autoFields($autoFieldsBefore);
```

Comment:
I think a better solution might be to reset the eagerLoader instance in the `cleanCopy()` method.